src/main/java/com/pkb/unit/Unit.java

```diff
@@ -36,7 +37,9 @@
 import io.vavr.collection.List;
 
 /**
- * A Unit is a base class for a restartable,
+ * A Unit is a base class for a restartable, stoppable service that can have depencencies and
+ * that other Unitscan depend on. When a Unit is started then it tries the start its dependencies.
+ * Once all of its dependencies has been started it starts itself.
```

Comment:
Please add 
```
A unit may also have a desired state (enabled or disabled), in which case it will try periodically to return to the started (or stopped) state.
```